### PR TITLE
FEAT: Add support for multi-selection in the DataFrameEditor.

### PR DESCRIPTION
**Pull request by jonathanrocher**
Tiny change to add a `multi_select` attribute on the `DataFrameEditor` factory, and to pass it to the underlying `TabularEditor`. The attribute defaults to `False` to match the `TabularEditor` behavior.

### CODE REVIEW

File: traitsui/tests/ui_editors/test_data_frame_editor.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 try:
     from pandas import DataFrame
 except ImportError as exc:
-    print "Can't import Pandas: skipping"
+    print("Can't import Pandas: skipping")
```
Comment (jonathanrocher) [inline traitsui/tests/ui_editors/test_data_frame_editor.py]: Unrelated Py3 fix.

**Comment by jonathanrocher** [comment]
Thanks for the review. I will resolve conflicts, add an entry in the changelog, and merge.

**Comment by corranwebster** [comment]
Merging - appveyor problems should have been fixed by #420.
